opsgenie/resource_opsgenie_team_membership.go

```diff
@@ -0,0 +1,163 @@
+package opsgenie
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/team"
+	"log"
+	"strings"
+)
+
+func resourceOpsGenieTeamMembership() *schema.Resource { //TODO encode the e-mail addrs (e.g. because of +)? https://github.com/opsgenie/opsgenie-go-sdk-v2/issues/62
+	return &schema.Resource{
+		Create: resourceOpsGenieTeamMembershipCreate,
+		Read:   handleNonExistentResource(resourceOpsGenieTeamMembershipRead),
+		//Update: resourceOpsGenieTeamMembershipUpdate, // requires https://github.com/opsgenie/opsgenie-go-sdk-v2/issues/59
+		Delete: resourceOpsGenieTeamMembershipDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"role": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "user",
+				ForceNew: true,
+			},
+			"team_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceOpsGenieTeamMembershipCreate(d *schema.ResourceData, meta interface{}) error {
+
+	userID := d.Get("user_id").(string)
+	role := d.Get("role").(string)
+	teamID := d.Get("team_id").(string)
+
+	log.Printf("[INFO] Adding user %q to team %q", teamID, userID)
+
+	client, err := team.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+
+	// add member to team
+	_, err = client.AddMember(context.Background(), &team.AddTeamMemberRequest{
+		TeamIdentifierType:  team.Id,
+		TeamIdentifierValue: teamID,
+		User: team.User{
+			ID: userID,
+		},
+		Role: role, //TODO lowercase?
```

Comment:
I did not really understand why (global) user and team roles have different casing. Should we do some user input sanitization here?

Reply:
Customers can create a custom role for their OG accounts. But why do you think we use lowercase

Reply:
standard team roles are lowercase. System user roles start with an uppercase. A little bit of a mess... :D

Reply:
will remove the comment in a min